### PR TITLE
Add experimental feature to import record data from csv file

### PR DESCRIPTION
**Pull request by modelrailroader**
#2163

### CODE REVIEW

File: phpmyfaq/admin/index.php [hunk 1]
```diff
@@ -72,10 +71,10 @@
 //
 try {
     Translation::create()
-        ->setLanguagesDir(PMF_TRANSLATION_DIR)
-        ->setDefaultLanguage('en')
-        ->setCurrentLanguage($faqLangCode)
-        ->setMultiByteLanguage();
+            ->setLanguagesDir(PMF_TRANSLATION_DIR)
```
Comment (thorsten) [inline phpmyfaq/admin/index.php:74]: The indent here is wrong :-)

File: phpmyfaq/src/phpMyFAQ/Faq/FaqImport.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Class FaqImport
+ *
+ * @package phpMyFAQ\Faq
+ */
+class FaqImport
+{
+    public function parseCSV($handle): array
+    {
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Faq/FaqImport.php:28]: We need to validate the CSV

File: phpmyfaq/admin/csv.import.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+if ($user->perm->hasPermission($user->getUserId(), 'add_faq')) {
+    $twig = new TwigWrapper(PMF_ROOT_DIR . '/assets/templates');
+    $template = $twig->loadTemplate('./admin/content/csv.import.twig');
+    $faqImport = new FaqImport();
+    $templateVars = [];
+
+    if (isset($_FILES['userfile']) && 0 === $_FILES['userfile']['error'] && $faqImport->isCSVFile($_FILES['userfile'])) {
+        $handle = fopen($_FILES['userfile']['tmp_name'], 'r');
+        $csvData = $faqImport->parseCSV($handle);
+        foreach ($csvData as $record) {
```
Comment (thorsten) [inline phpmyfaq/admin/csv.import.php:52]: This logic should be in the FaqImport class as well

**Review by thorsten — CHANGES_REQUESTED** [review]

**Comment by modelrailroader** [comment]
Close due to the necessary to create a new PR #2800

**Pull request closed without merging**